### PR TITLE
Move documentation to Sphinx

### PR DESCRIPTION
**Pull request by astrofrog**
This also includes a docs build on CircleCI with a link provided by Giles

### CODE REVIEW

**Comment by astrofrog** [comment]
Just confirming this also works properly on RTD: https://astropy-helpers.readthedocs.io/en/latest/

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+
+project = 'astropy-helpers'
+copyright = '2019, The Astropy Developers'
```
Comment (bsipocz) [inline docs/conf.py]: what's the policy on the year here? I mean it keeps coming up every year, and I keep forgetting. Should it be the full time period, or the latest or the first year?

**Comment by bsipocz** [comment]
I liked the readme way of documenting infrastructure like the helpers, and also not a big fan of circleCI, but agreed that the readme was already too long.

**Comment by astrofrog** [comment]
The reason I did this is because I needed to add more docs and this was getting very long. I'm going to add some of the high-level description back to the README though re: @eteq's comment.

File: docs/conf.py (hunk 1 above)
Comment (astrofrog) [reply docs/conf.py]: For now I'll fix it to make it match the license, but we can update this if we figure out what the right one is.

Comment (bsipocz) [reply docs/conf.py]: Oh, sure. This was more like self mumbling that we need to figure out and act on separately across the whole ecosystem. (preferably with batchpr? 😉  )

Comment (astrofrog) [reply docs/conf.py]: Ok cool :)

**Comment by astrofrog** [comment]
I've backported (and updated) this to v3.1.1

**Pull request merged**